x/ibc/core/02-client/client/cli/tx.go

```diff
@@ -13,54 +12,58 @@ import (
+		Long: `create a new IBC client with the specified client state and consensus state
+	- ClientState JSON example: {"@type":"/ibc.lightclients.solomachine.v1.ClientState","sequence":"1","frozen_sequence":"0","consensus_state":{"public_key":{"@type":"/cosmos.crypto.secp256k1.PubKey","key":"AtK50+5pJOoaa04qqAqrnyAqsYrwrR/INnA6UPIaYZlp"},"diversifier":"testing","timestamp":"10"},"allow_update_after_proposal":false}
+	- ConsensusState JSON example: {"@type":"/ibc.lightclients.solomachine.v1.ConsensusState","public_key":{"@type":"/cosmos.crypto.secp256k1.PubKey","key":"AtK50+5pJOoaa04qqAqrnyAqsYrwrR/INnA6UPIaYZlp"},"diversifier":"testing","timestamp":"10"}`,
+		Example: fmt.Sprintf("%s tx ibc %s create [path/to/client_state.json] [path/to/consensus_state.json] --from node0 --home ../node0/<app>cli --chain-id $CID", version.AppName, types.SubModuleName),
 		Args:    cobra.ExactArgs(2),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			clientCtx, err := client.GetClientTxContext(cmd)
 			if err != nil {
 				return err
 			}
-			sequence, err := strconv.ParseUint(args[0], 10, 64)
-			if err != nil {
-				return err
-			}
-
 			cdc := codec.NewProtoCodec(clientCtx.InterfaceRegistry)
 
+			// attempt to unmarshal client state argument
+			var clientState exported.ClientState
+			if err := cdc.UnmarshalInterfaceJSON([]byte(args[0]), &clientState); err != nil {
+
+				// check for file path if JSON input is not provided
+				contents, err := ioutil.ReadFile(args[0])
+				if err != nil {
+					return errors.Wrap(err, "neither JSON input nor path to .json file for client state were provided")
+				}
+
+				if err := cdc.UnmarshalInterfaceJSON(contents, &clientState); err != nil {
+					return errors.Wrap(err, "error unmarshalling client state file")
+				}
+			}
+
 			// attempt to unmarshal consensus state argument
-			consensusState := &types.ConsensusState{}
-			if err := cdc.UnmarshalJSON([]byte(args[1]), consensusState); err != nil {
+			var consensusState exported.ConsensusState
+			if err := cdc.UnmarshalInterfaceJSON([]byte(args[1]), &consensusState); err != nil {
 
 				// check for file path if JSON input is not provided
 				contents, err := ioutil.ReadFile(args[1])
 				if err != nil {
 					return errors.Wrap(err, "neither JSON input nor path to .json file for consensus state were provided")
 				}
 
-				if err := cdc.UnmarshalJSON(contents, consensusState); err != nil {
+				if err := cdc.UnmarshalInterfaceJSON(contents, &consensusState); err != nil {
 					return errors.Wrap(err, "error unmarshalling consensus state file")
 				}
 			}
 
-			allowUpdateAfterProposal, _ := cmd.Flags().GetBool(flagAllowUpdateAfterProposal)
-
-			clientState := types.NewClientState(sequence, consensusState, allowUpdateAfterProposal)
-			msg, err := clienttypes.NewMsgCreateClient(clientState, consensusState, clientCtx.GetFromAddress())
+			msg, err := types.NewMsgCreateClient(clientState, consensusState, clientCtx.GetFromAddress())
```

Comment:
Note: I intentionally do not return the client identifier generated since the call below is `GenerateOrBroadcastTx`. This code cannot return a client identifier if the transaction is simply generated and not broadcasted. If users want this functionality, they should use external resources like a relayer. This should really only be for testing